dot/parachain/candidate-validation/worker.go

```diff
@@ -0,0 +1,93 @@
+package candidatevalidation
+
+import (
+	parachainruntime "github.com/ChainSafe/gossamer/dot/parachain/runtime"
+	parachaintypes "github.com/ChainSafe/gossamer/dot/parachain/types"
+)
+
+// TODO(ed): figure out a better name for this that describes what it does
+type worker struct {
+	workerID parachaintypes.ValidationCodeHash
+	instance *parachainruntime.Instance
+	// TODO(ed): determine if wasProcessed is stored here or in host
+	isProcessed map[parachaintypes.CandidateHash]struct{}
+}
+
+type workerTask struct {
+	work             parachainruntime.ValidationParameters
+	maxPoVSize       uint32
+	candidateReceipt *parachaintypes.CandidateReceipt
+}
+
+func newWorker(validationCode parachaintypes.ValidationCode) (*worker, error) {
+	validationRuntime, err := parachainruntime.SetupVM(validationCode)
+
+	if err != nil {
+		return nil, err
+	}
+	return &worker{
+		workerID: validationCode.Hash(),
+		instance: validationRuntime,
+	}, nil
+}
+
+func (w *worker) executeRequest(task *workerTask) (*ValidationResult, error) {
+	logger.Debugf("[EXECUTING] worker %x task %v", w.workerID, task.work)
+	candidateHash, err := parachaintypes.GetCandidateHash(task.candidateReceipt)
+	if err != nil {
+		return nil, err
+	}
+
+	// do isProcessed check here
+	if _, ok := w.isProcessed[candidateHash]; ok {
+		// TODO: determine what the isPreccessed check should return, and if re-trying is allowed
+		//  get a better understanding of what the isProcessed check should be checking for
+		logger.Debugf("candidate %x already processed", candidateHash)
```

Comment:
should we return here?

Reply:
Yes, I've updated this to return the cached value.